File: group_anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        anagrams_seen = {}

        for x in strs:
            sorted_word = ''.join(sorted(x))
            if sorted_word in anagrams_seen:
                anagrams_seen[sorted_word].append(x)
            else:
                anagrams_seen[sorted_word] = [x]

        return anagrams_seen.values() 

File: group_anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function (strs) {
    let hash = {};

    for (let i = 0; i < strs.length; i++) {
        let sorted = strs[i].split('').sort().join('');

        if (hash[sorted] == null) {
            hash[sorted] = [strs[i]];
        } else {
            hash[sorted].push(strs[i]);
        };
    };

    let anagrams = Object.values(hash);

    return anagrams;
};